docs/data/data-grid/pivoting/pivoting.md

```diff
@@ -2,16 +2,170 @@
+    sort?: 'asc' | 'desc';
+    hidden?: boolean;
+  }>;
+  values: Array<{
+    field: string;
+    aggFunc: string;
+    hidden?: boolean;
+  }>;
+}
+```
+
+### Initialize pivoting
+
+Use the `initialState` prop to initialize uncontrolled pivoting.
+This is the recommended method unless you specifically need to control the state.
+You can initialize the pivot model, toggle pivot panel visibility, and toggle the pivot mode as shown below:
+
+```tsx
+<DataGridPremium
+  initialState={{ pivoting: { model: pivotModel, active: true, panelOpen: true } }}
+/>
+```
+
+{{"demo": "GridPivotingInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Controlled pivoting
+
+For fully controlled pivoting state, you can use the following props:
+
+- Pivot model:
+  - `pivotModel`: Controls the current pivot configuration.
+  - `onPivotModelChange`: Callback fired when the pivot model changes.
+- Pivot mode toggle:
+  - `pivotActive`: Controls whether pivot mode is active.
+  - `onPivotActiveChange`: Callback fired when the pivot mode changes between active and inactive.
+- Pivot panel:
+  - `pivotPanelOpen`: Controls whether the pivot panel is open.
+  - `onPivotPanelOpenChange`: Callback fired when the pivot panel is opened or closed.
+
+{{"demo": "GridPivotingControlled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Using fields in the pivot model multiple times
+
+While this is not supported yet, we are working to bring this feature to the Data Grid.
+
+## Disable pivoting
+
+To disable pivoting, set the `disablePivoting` prop to `true`:
+
+```tsx
+<DataGridPremium disablePivoting />
+```
+
+## Derived columns in pivot mode
+
+In pivot mode, it's often useful to group data by a year or quarter.
+The Data Grid automatically generates year and quarter columns for each **Date** column for this purpose.
+
+In the demo below, the **Transaction Date** column is represented by additional columns in pivot mode: **Transaction Date (Year)** and **Transaction Date (Quarter)**:
+
+{{"demo": "GridPivotingFinancial.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom derived columns
+
+Use the `getPivotDerivedColumns` prop to customize derived columns.
+This prop is called for each original column and returns an array of derived columns, or `undefined` if no derived columns are needed.
+
+{{"demo": "GridGetPivotDerivedColumns.js", "bg": "inline", "defaultCodeOpen": true}}
+
+## Sticky column groups
+
+Depending on the pivot mode, some column groups might exceed the width of the Data Grid viewport.
+To improve the user experience, you can make these column groups "sticky" so that the column group labels remain visible while scrolling horizontally.
+
+The demo below illustrates how to do this using the `sx` prop to apply the necessary styles:
+
+{{"demo": "GridPivotingMovies.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Advanced demo
+
+{{"demo": "GridPivotingCommodities.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
I loved that the Rows column is sticky in this demo, and I'd say most use cases will likely benefit from that.
I believe we should have a follow up issue to make this an out-of-the-box feature for pivoting. (obviously with ways to turn it off).